### DISCUSSION TITLE
Don't we need to add codes to handle a server-side response when we use turnstile?

### DISCUSSION THREAD

**Discussion by matskohe** [supabase/supabase#19291] [Questions]
I am adding turnstile to my supabase auth signup with the module react-turnstile by following the official instruction [Enable Captcha Protection](https://supabase.com/docs/guides/auth/auth-captcha?captcha-method=turnstile-1). I noticed that Cloudflare Turnstile recommends [validating the server-side response](https://developers.cloudflare.com/turnstile/get-started/#validate-the-server-side-response) and the react-turnstile has a section [Validating a token](https://docs.page/marsidev/react-turnstile/validating-a-token). Is it better to do that by following react-turnstile "Validating a token" page?

**Comment by matskohe** [comment]
The Error is returned when I login with the token which is server side validated.
The error below
```
[AuthApiError: captcha protection: request disallowed (timeout-or-duplicate)] {
  __isAuthError: true,
  name: 'AuthApiError',
  status: 400
```
Maybe server-side validation is done by supabase system, so you must not do it by yourself?

**Comment by muhaimincs** [comment]
Yes how do we tackle this seamlessly
